modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -141,42 +155,48 @@ const DomParser = (settings?: DomParserSettings, schema = Schema()): DomParser =
 
       // Found a suitable parent
       if (parent && parents.length > 1) {
-        // Reverse the array since it makes looping easier
-        parents.reverse();
-
-        // Clone the related parent and insert that after the moved node
-        newParent = filterNode(parents[0].clone());
-
-        // Start cloning and moving children on the left side of the target node
-        let currentNode = newParent;
-        for (let i = 0; i < parents.length - 1; i++) {
-          if (schema.isValidChild(currentNode.name, parents[i].name)) {
-            tempNode = filterNode(parents[i].clone());
-            currentNode.append(tempNode);
-          } else {
-            tempNode = currentNode;
-          }
+        // If the node won't be a valid child of the parent then just unwrap
```

Comment:
Just re: understanding this, it lost me for a second that this comment is more about the else case. Could we word it more like "If it's a valid child then try to move it, else just unwrap it"? The comment reflects the logic a bit better that way.